src/tools/compiletest/src/runtest.rs

```diff
@@ -1118,31 +1120,48 @@ impl<'test> TestCx<'test> {
 
     fn build_all_auxiliary(&self, of: &TestPaths, aux_dir: &Path, rustc: &mut Command) {
         for rel_ab in &self.props.aux.builds {
-            self.build_auxiliary(of, rel_ab, &aux_dir, false /* is_bin */);
+            self.build_auxiliary(of, rel_ab, &aux_dir, None);
         }
 
         for rel_ab in &self.props.aux.bins {
-            self.build_auxiliary(of, rel_ab, &aux_dir, true /* is_bin */);
+            self.build_auxiliary(of, rel_ab, &aux_dir, Some(AuxType::Bin));
         }
 
+        let path_to_crate_name = |path: &str| -> String {
+            path.rsplit_once('/')
```

Comment:
Suggestion: I can't remember if this `path` can be a Windows path where the path separator can be different. Maybe operate on a `Path` instead (I should convert compiletest to use `camino` and just use `Utf8Path`s everywhere), and use `Path::file_stem` (compiletest has an assertion that test filenames cannot include `.`s).

Reply:
I don't think we support backslash in test headers, otherwise they wouldn't work at all on unix. It would require changes throughout to support that.

Reply:
Oh right, good point, nevermind.